libminifi/src/provenance/ProvenanceRepository.cpp

```diff
@@ -36,6 +36,7 @@ void ProvenanceRepository::run() {
     uint64_t curTime = getTimeMillis();
     uint64_t size = repoSize();
     if (size >= purgeThreshold) {
+      std::lock_guard<std::mutex> lock(mutex_);
```

Comment:
yes, levelDB is thread save, but when we purge the record in provenanceRepository task, i do not want to ProvenanceReportTask to pick the record and ship to NiFi.